compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/NodeFinder.java

```diff
@@ -0,0 +1,1484 @@
+        lookupNode(bLangArrowFunction.body);
+    }
+
+    @Override
+    public void visit(BLangIntRangeExpression intRangeExpression) {
+        if (setEnclosingNode(intRangeExpression, intRangeExpression.pos)) {
+            return;
+        }
+
+        lookupNode(intRangeExpression.startExpr);
+        lookupNode(intRangeExpression.endExpr);
+    }
+
+    @Override
+    public void visit(BLangRestArgsExpression bLangVarArgsExpression) {
+        if (setEnclosingNode(bLangVarArgsExpression, bLangVarArgsExpression.pos)) {
+            return;
+        }
+
+        lookupNode(bLangVarArgsExpression.expr);
+    }
+
+    @Override
+    public void visit(BLangNamedArgsExpression bLangNamedArgsExpression) {
+        if (setEnclosingNode(bLangNamedArgsExpression, bLangNamedArgsExpression.pos)) {
+            return;
+        }
+
+        lookupNode(bLangNamedArgsExpression.expr);
+    }
+
+    @Override
+    public void visit(BLangIsAssignableExpr assignableExpr) {
+        if (setEnclosingNode(assignableExpr, assignableExpr.pos)) {
+            return;
+        }
+
+        lookupNode(assignableExpr.lhsExpr);
+        lookupNode(assignableExpr.typeNode);
+    }
+
+    @Override
+    public void visit(BLangCheckedExpr checkedExpr) {
+        if (setEnclosingNode(checkedExpr, checkedExpr.pos)) {
+            return;
+        }
+
+        lookupNode(checkedExpr.expr);
+    }
+
+    @Override
+    public void visit(BLangFail failExpr) {
+        if (setEnclosingNode(failExpr, failExpr.pos)) {
+            return;
+        }
+
+        lookupNode(failExpr.expr);
+    }
+
+    @Override
+    public void visit(BLangCheckPanickedExpr checkPanickedExpr) {
+        if (setEnclosingNode(checkPanickedExpr, checkPanickedExpr.pos)) {
+            return;
+        }
+
+        lookupNode(checkPanickedExpr.expr);
+    }
+
+    @Override
+    public void visit(BLangServiceConstructorExpr serviceConstructorExpr) {
+        if (setEnclosingNode(serviceConstructorExpr, serviceConstructorExpr.pos)) {
+            return;
+        }
+
+        lookupNode(serviceConstructorExpr.serviceNode);
+    }
+
+    @Override
+    public void visit(BLangTypeTestExpr typeTestExpr) {
+        lookupNode(typeTestExpr.expr);
+        lookupNode(typeTestExpr.typeNode);
+    }
+
+    @Override
+    public void visit(BLangIsLikeExpr typeTestExpr) {
+        lookupNode(typeTestExpr.expr);
+        lookupNode(typeTestExpr.typeNode);
+    }
+
+    @Override
+    public void visit(BLangIgnoreExpr ignoreExpr) {
+        // ignore
+    }
+
+    @Override
+    public void visit(BLangAnnotAccessExpr annotAccessExpr) {
+        setEnclosingNode(annotAccessExpr, annotAccessExpr.pos);
+    }
+
+    @Override
+    public void visit(BLangQueryExpr queryExpr) {
+        if (setEnclosingNode(queryExpr, queryExpr.pos)) {
+            return;
+        }
+
+        lookupNodes(queryExpr.queryClauseList);
+    }
+
+    @Override
+    public void visit(BLangTableMultiKeyExpr tableMultiKeyExpr) {
```

Comment:
Calling the super methods throws assertion error

Reply:
Was intentional, since these are not yet supported. Throwing an unsupported expression exception would be more appropriate i guess?

Reply:
This will be OK during the development until we finish the implementation. Otherwise these might cause broken usages even for valid cases. We had such issues with earlier LS visitor implementations

Reply:
@SupunS yeah, we override the base expectation